From 5c8f42c67d94bc4347401e49d6bc2895662f3333 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@freescale.com>
Date: Wed, 9 Dec 2015 18:51:22 +0800
Subject: [PATCH 1213/1691] MLK-11985-1 ARM: imx: fix coverity scan issues

commit 5c8f42c67d94bc4347401e49d6bc2895662f3333 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

1. Add return value check to fix below warning:

check_return: Calling of_property_read_u32 without
checking return value.

2. Improve code to fix below warning:

copy_paste_error: i in of_clk_get(dev->of_node, i)
looks like a copy-paste error.

Signed-off-by: Anson Huang <Anson.Huang@freescale.com>
---
 arch/arm/mach-imx/gpc.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/gpc.c b/arch/arm/mach-imx/gpc.c
index bda6385..1814dac 100644
--- a/arch/arm/mach-imx/gpc.c
+++ b/arch/arm/mach-imx/gpc.c
@@ -750,12 +750,12 @@ static int imx_gpc_genpd_init(struct device *dev, struct regulator *pu_reg)
 	ipg = of_clk_get(dev->of_node, pu_clks);
 
 	/* Get disp domain clks */
-	for (k = 0, i = pu_clks + ipg_clks; i < pu_clks + ipg_clks + disp_clks;
-		i++, k++) {
+	for (i = pu_clks + ipg_clks; i < pu_clks + ipg_clks + disp_clks;
+		i++) {
 		clk = of_clk_get(dev->of_node, i);
 		if (IS_ERR(clk))
 			break;
-		imx6s_display_domain.clk[k] = clk;
+		imx6s_display_domain.clk[k++] = clk;
 	}
 	imx6s_display_domain.num_clks = k;
 
@@ -797,7 +797,9 @@ static int imx_gpc_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	of_property_read_u32(pdev->dev.of_node, "fsl,ldo-bypass", &bypass);
+	if (of_property_read_u32(pdev->dev.of_node, "fsl,ldo-bypass", &bypass))
+		dev_warn(&pdev->dev,
+			"no fsl,ldo-bypass found!\n");
 	/* We only bypass pu since arm and soc has been set in u-boot */
 	if (pu_reg && bypass)
 		regulator_allow_bypass(pu_reg, true);
-- 
1.9.1

